pkg/proxy/util/utils.go

```diff
@@ -209,9 +209,9 @@ func GetNodeAddresses(cidrs []string, nw NetworkInterfacer) (sets.String, error)
 		}
```

Comment:
Once a ZeroCIDR has been inserted the `uniqueAddressList.Has(IPvXZeroCIDR)` later on will _always_ be true and thus prevent _any_ other ip from beeing inserted. So you should return here. Note that the nodePortAddresses have been filtered so there is no chance that an IPv6ZeroCIDR is passed for the IPv4 proxier.

Reply:
(but there are unit-tests that tries mix ipv4/6 this even though it will never happen)